gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1268 @@
+                grouped_by_labels[label].append(topic)
+
+        # then aggregate their amount_parent_labels by maxing and parent_labels by concatenating
+        # the result is sorted by amount_parent_labels and stored in sorted_clusters
+        sorted_clusters = []
+        for label, group in grouped_by_labels.items():
+            amount_parent_labels = 0
+            parent_labels = []  # will be a list of sets
+            for topic in group:
+                amount_parent_labels = max(topic["amount_parent_labels"], amount_parent_labels)
+                parent_labels.append(topic["parent_labels"])
+            # - here, nan means "not yet evaluated"
+            # - removing empty sets from parent_labels
+            sorted_clusters.append({
+                "amount_parent_labels": amount_parent_labels,
+                "parent_labels": [x for x in parent_labels if len(x) > 0],
+                "is_valid": np.nan,
+                "label": label
+            })
+
+        # start with the most significant core.
+        sorted_clusters = sorted(sorted_clusters,
+            key=lambda cluster: (
+                cluster["amount_parent_labels"],
+                cluster["label"]  # makes sorting deterministic
+            ), reverse=True)
+
+        def remove_from_all_sets(label):
+            """removes a label from every set in "parent_labels" for
+            each core in sorted_clusters."""
+            for a in sorted_clusters:
+                if label in a["parent_labels"]:
+                    a["parent_labels"].remove(label)
+
+        # APPLYING THE RULES 1 to 3
+        # iterate over the cluster labels, see which clusters/labels
+        # are valid to cause the creation of a stable topic
+        for i, cluster in enumerate(sorted_clusters):
+            label = cluster["label"]
+            # label is iterating over 0, 1, 3, ...
+
+            # 1. rule - remove if the cores in the cluster have no parents
+            if cluster["amount_parent_labels"] == 0:
+                # label is NOT VALID
+                cluster["is_valid"] = False
+                remove_from_all_sets(label)
+
+            # 2. rule - remove if it has less than min_cores as parents
+            # (parents are always also cores)
+            if len(cluster["parent_labels"]) < min_cores:
+                cluster["is_valid"] = False
+                remove_from_all_sets(label)
+
+            # 3. checking for "easy_valid"s
+            # checks if the core has at least min_cores of cores with the only label as itself
+            if cluster["amount_parent_labels"] >= 1:
+                if sum(map(lambda x: x == {label}, cluster["parent_labels"])) >= min_cores:
+                    cluster["is_valid"] = True
+
+        # Reaplying the rule 3
+        # this happens when we have a close relationship among 2 or more clusters
+        for cluster in [cluster for cluster in sorted_clusters if np.isnan(cluster["is_valid"])]:
```

Comment:
Avoid unnecessarily overloading variable names:

```suggestion
        for cluster in [c for c in sorted_clusters if np.isnan(c["is_valid"])]:
```

Reply:
Done